fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetadata.java

```diff
@@ -0,0 +1,146 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.connector.jdbc;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.JDBCResource;
+import com.starrocks.catalog.JDBCTable;
+import com.starrocks.catalog.Table;
+import com.starrocks.common.DdlException;
+import com.starrocks.connector.ConnectorMetadata;
+import com.starrocks.external.HiveMetaStoreTableUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class JDBCMetadata implements ConnectorMetadata {
+
+    private static Logger LOG = LogManager.getLogger(JDBCMetadata.class);
+
+    private Map<String, String> properties;
+    private String dirverType;
+
+    public JDBCMetadata(Map<String, String> properties) throws DdlException {
+        this.properties = properties;
+        try {
+            Class.forName(properties.get(JDBCResource.DRIVER_CLASS));
+        } catch (ClassNotFoundException e) {
+            LOG.warn(e.getMessage());
+            throw new DdlException("doesn't find class: " + e.getMessage());
+        }
+        if (properties.get(JDBCResource.DRIVER_CLASS).toLowerCase().contains("mysql")) {
+            dirverType = "mysql";
+        } else {
+            LOG.warn("not support yet");
+            throw new DdlException("not support yet");
+        }
+    }
+
+    @Override
+    public List<String> listDbNames() throws DdlException {
+        try (Connection connection = DriverManager.getConnection(properties.get(JDBCResource.URI),
```

Comment:
you might wanna extract this as a method since it's there are 3 similar duplication